### PR TITLE
Improve build process and tooling

### PR DESCRIPTION
**Pull request by gtramontina**
There's a lot going on on this PR, everything related to building and linting. I'm more than happy to break this apart into smaller PRs if you find it useful.
- The `eslint` rules applied are the recommended ones by `eslint` itself plus a few overrides.
- Added `.editorconfig` so everyone has their editors "configured" similarly.
- I've also added `nyc` to report code coverage. We might also want to integrate it with [codeclimate](https://codeclimate.com/) or [coveralls](https://coveralls.io/) for some neat charts and details…
- As far as building, the `dist/` directory does not need to exist anymore, given that it is going to be built on the `prepublish` npm hook. We can take a step further and completely automate the releasing/publishing process with [semantic-release](https://github.com/semantic-release/semantic-release). It requires adding GitHub's and NPM's keys to Travis and following a commit message standard (just like the one I used on this PR).

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -1 +1,8 @@
-{ "presets": ["es2015"] }
+{
+  "presets": [ "es2015" ],
+  "env": {
+    "test": {
+      "plugins": [ "istanbul" ]
```
Comment (vvgomes) [inline .babelrc:5]: Istanbul is cool!

File: package.json [hunk 2]
```diff
@@ -16,17 +16,19 @@
...
-      "es2015"
-    ]
-  },
   "scripts": {
-    "test": "mocha --compilers js:babel-register --recursive",
-    "dist": "mkdir -p dist && rm -rf dist/* && babel lib --out-dir dist"
+    "lint": "eslint {lib,test}/**/*",
+    "test": "nyc mocha --compilers js:babel-register --recursive",
+    "dist": "babel lib/ --out-dir dist/",
+    "prepublish": "npm test && npm run lint && npm run dist"
```
Comment (vvgomes) [inline package.json:32]: Do we need to explicitly run it before publishing or is it a npm hook?

File: test/aggregate.test.js [hunk 3]
```diff
@@ -12,10 +12,10 @@ describe("Aggregate{}", () => {
   const config = {
     initialState: 0,
     eventHandlers: {
-      counterIncremented: (state, event) => state + 1
+      counterIncremented: (state) => state + 1
     },
     commandHandlers: {
-      incrementCounter: (state, command) => Success([{ type: "counterIncremented" }])
+      incrementCounter: () => Success([{ type: "counterIncremented" }])
     }
```
Comment (vvgomes) [inline test/aggregate.test.js:19]: I see the point of removing unused args. However, do you think they could help as documentation? The arguments here can be a good hint for users about the handlers interface.

File: test/bravent.test.js [hunk 4]
```diff
@@ -6,12 +6,12 @@ describe("a Bravent aggregate", () => {
...
-      counterDecremented: (state, event) => state - 1
+      counterIncremented: (state) => state + 1,
+      counterDecremented: (state) => state - 1
     },
     commandHandlers: {
-      incrementCounter: (state, command) => [{ type: "counterIncremented" }],
-      decrementCounter: (state, command) => [{ type: "counterDecremented" }]
+      incrementCounter: () => [{ type: "counterIncremented" }],
+      decrementCounter: () => [{ type: "counterDecremented" }]
     }
```
Comment (vvgomes) [inline test/bravent.test.js:15]: I'd say here the same thing about documentation as mentioned previously.

**Review by vvgomes — APPROVED** [review]
Take a look at the questions, please 👍

File: package.json (hunk 2 above)
Comment (gtramontina) [reply package.json:32]: It runs whenever we run `npm publish` or `npm install` (locally, without arguments – `npm install foo-module` won't trigger it) [[source]](https://docs.npmjs.com/misc/scripts).

Whenever you have some time, I'd suggest to setup [semantic-release](https://github.com/semantic-release/semantic-release), it makes releasing and publishing a breeze. 😉

File: test/aggregate.test.js (hunk 3 above)
Comment (gtramontina) [reply test/aggregate.test.js:19]: That's a good point. I just went blindly removing unused args. How about ignoring arguments that match [`^_`](http://eslint.org/docs/rules/no-unused-vars#argsignorepattern) (just like rubocop does in when enforcing the ruby style-guide)?

File: test/bravent.test.js (hunk 4 above)
Comment (gtramontina) [reply test/bravent.test.js:15]: ☝️

**Pull request merged**